docs/docs/tutorials/agents.md

```diff
@@ -1111,6 +1111,280 @@ String result = (String) bankSupervisor.invoke(input);
 
 If both are provided, the invocation value overrides the build-time `supervisorContext`.
 
+## Custom agentic patterns
+
+The agentic patterns discussed so far are provided out-of-the-box by the `langchain4j-agentic` module, but what if none of them fit the specific needs of your application? In this case it is possible to create your own custom pattern, that orchestrates the interactions among a set of subagents in a way that is tailored to your requirements.
+
+In more details an agentic pattern is simply the specification of an execution plan for the subagents that it coordinates. This plan can be defined by implementing the following `Planner` interface:
+
+```java
+public interface Planner {
+
+    default void init(AgenticScope agenticScope, AgentInstance plannerAgent, List<AgentInstance> subagents) { }
+
+    default Action firstAction(AgenticScope agenticScope) {
+        return nextAction(agenticScope, null);
+    }
+
+    Action nextAction(AgenticScope agenticScope, AgentExecution lastAgentExecution);
```

Comment:
Shouldn't `lastAgentExecution` be called `previousAgentExecution`?

Reply:
Good point, renamed.